pytrip/vdx.py

```diff
@@ -204,12 +210,43 @@ def voi_names(self):
         names = [voi.name for voi in self.vois]
         return names
 
-    def add_voi(self, voi):
+    def add_voi(self, voi, added_manually: bool = False) -> None:
         """ Appends a new voi to this class.
+        If added_manually flag is True, sets voi color as one from spare ones.
+        If there is no spare colors, calls assign_voi_colors
 
         :param Voi voi: the voi to be appended to this class.
+        :param bool added_manually: flag that changes way VOI is added, default False
+        """
+        if added_manually:
+            if len(self._spare_voi_colors) > 0:
+                color = self._spare_voi_colors.pop()
+                voi.set_color(color)
+                self.vois.append(voi)
+            else:
+                self.assign_voi_colors(k=6)
+        else:
+            self.vois.append(voi)
+
+    def assign_voi_colors(self, k: int = 3) -> None:
```

Comment:
Avoid type hinting in pure pytrip.  pytrip still supports python 2.7 and some ancient 3.x versions.

Reply:
Removed